modules/net-vpc/subnets.tf

```diff
@@ -33,7 +33,7 @@ locals {
       secondary_ip_ranges   = try(v.secondary_ip_ranges, null)
       iam                   = try(v.iam, [])
       iam_members           = try(v.iam_members, [])
-      purpose               = try(v.purpose, null)
+      purpose               = try(v.global, null)
```

Comment:
This part refers to YAML file parsing, for backward compatibility please keep `v.purpose` here.

As you are adding `purpose: GLOBAL_MANAGED_PROXY` you need to add specific handling below, when creating `subnets_proxy_only` from `local._factory_subnets`. You need to add `active` true/false depending on type of what is the purpose (global or regional proxy)